sqlmesh/core/engine_adapter/mixins.py

```diff
@@ -227,6 +227,26 @@ def _truncate_table(self, table_name: TableName) -> None:
 
 
 class VarcharSizeWorkaroundMixin(EngineAdapter):
+    def _default_precision_to_max(
+        self, columns_to_types: t.Dict[str, exp.DataType]
+    ) -> t.Dict[str, exp.DataType]:
+        # get default lengths for types that support "max" length
+        types_with_max_default_param = {
+            k: [self.SCHEMA_DIFFER.parameterized_type_defaults[k][0][0]]
```

Comment:
I'm not sure we care about all types. I think we only care about `Varchar` don't we?

Reply:
Wouldn't this simplify the implementation?

Reply:
Or do we think the same issue applies to types `VARBINARY`?

Reply:
I figured it could happen for any of these types due to transpilation, but VARCHAR is much more widely used than NVARCHAR/CHAR/VARBINARY so in practice it's probably the only one that matters. I'll update.